pages/c/[communityName].js

```diff
@@ -0,0 +1,48 @@
+import Constant from "utils/constant";
+import RedirectUtils from "utils/redirect"
+import UserAgentUtils from "utils/userAgent"
+import parser from 'ua-parser-js';
+import { useEffect } from "react"
+import { useRouter } from "next/router"
+
+export async function getServerSideProps(context) {
+    const { communityName } = context.query
+    let isDynamicLink = communityName?.indexOf('+') >= 0
+    let userAgent = parser(context?.req?.headers['user-agent'])
+    
+    if (UserAgentUtils.isMobile(userAgent)) {
+        let redirect = await RedirectUtils.redirectCommunityForMobileDevice(userAgent, communityName)
+        if (redirect && !isDynamicLink) return redirect
+    }
+
+    if(UserAgentUtils.isIos(userAgent)) {
+        return {
+            redirect: {
+                destination: Constant.Link.appstore,
+            }
+        }
+    }
+
+    if(UserAgentUtils.isAndroid(userAgent)) {
+        return {
+            redirect: {
+                destination: Constant.Link.playstore,
+            }
+        }
+    }
+
+    return {
+        redirect: {
+            destination: Constant.Link.bettersocial,
+        }
+    }
+}
+
+export default function CommunityPage({ communityName, isDynamicLink }) {
+    const router = useRouter()
+    useEffect(() => {
+        if (isDynamicLink) router.push(`/community/${communityName}`, undefined, { shallow: true })
+    }, [isDynamicLink])
+
+    return <>{communityName}</>
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `useEffect` hook, you're pushing a new route when `isDynamicLink` is true. However, there's no cleanup function provided. If the component unmounts before the routing operation completes, it could lead to memory leaks. Consider cancelling the routing operation in the cleanup function of the `useEffect` hook.


<!-- This is an auto-generated comment by OSS CodeRabbit -->